spec/ParseUser.spec.js

```diff
@@ -14,6 +14,24 @@ const Config = require('../lib/Config');
 const cryptoUtils = require('../lib/cryptoUtils');
 
 describe('Parse.User testing', () => {
+  it('retrieves original object when user logs in with third party auth', async done => {
+    let counter = 0;
+    Parse.Cloud.afterSave(Parse.User, async request => {
+      if (!request.original) {
+        const user = request.object;
+        if (user.get('authData') != null) {
+          if (counter > 0) {
+            done.fail();
```

Comment:
Can you simplify this code?

Form your issue description we should expect 2 things:
- that authData is not erased
- that the ParseUser.isNew is false

instead of `done.fail` you can use expect(user.get(...)).not.toBeUndefined();` and so on.

Reply:
The issue isn't only with the Android SDK, did try it also with the JS SDK (facebook login) same issue.
I get also the same error when using linkWith
`Unhandled promise rejection: TypeError: Parse.User.linkWith is not a function`
Can't tell which field is modified, i have tested it multiple times, what i noticed is only that sometimes the authData get erased and it doesn't retrieve the original object in afterSave, also note that the beforeSave is working correctly because it can detect when the user is new.